scripts/actions/check-job-progress.js

```diff
@@ -74,9 +74,11 @@ const main = async () => {
       .filter((batch) => batch && !batch.done)
       .map((batch) => batch.batchUid);
 
-    console.log(
-      `::set-output name=batchUids::${JSON.stringify(remainingBatches)}`
-    );
+    const outputBatchUids = remainingBatches.length
+      ? remainingBatches.join('[]')
+      : JSON.stringify(remainingBatches);
```

Comment:
Do you need this `JSON.stringify` if the `remainingBatches` are empty? Would just `[]` be sufficient here?

And out of curiosity, why use the `[]` delimiter here?

Reply:
There's some wonkiness with the `JSON.stringify` that I couldn't figure out. It wasn't able to parse the array when there was something in the array (got a Unknown JSON token error) but if I didn't have the JSON.stringify when there was no files in the array, then remainingbatches wouldn't get the correct number of arguments

Reply:
Thus why there's this ternary logic. It ensures that there is an empty array ('[]') or a delimiter for the remaining batches `batchuid1[]batchuid2`

Reply:
I guess what I mean is, if its empty, why even use `JSON.stringify` since you know its an empty array. Couldn't you just write the string `[]` itself? That being said, you may not even need this at all.

As a best practice, you'll want your values in both cases to be the same "type". Since your non-empty case is now a string delimited by a `[]`, you'll want your empty case to resemble this. Your empty case should no longer be an array, but rather a "string delimited by a []". In this case, the simplest from of an empty set of uids would just be an empty string.

This likely helps your parsing a bit too. Take the following 2 outputs:

```js
'[]'.split('[]') // => ["", ""]
''.split('[]') // => [""]
```

All of this being said, the `[]` delimiter is a bit strange. Instead, why not just use a comma delimiter? It looks like the array of values will be alphanumeric, so this should work great and won't feel quite as wonky. When doing so, `join`ing on an empty array gets you an empty string.

tl;dr; I would recommend simplifying this to the following:

```
const outputBatchUids = remainingBatches.join(',')

console.log(`::set-output name=batchUids::${outputBatchUids}`)
```